src/dos/cdrom.h

```diff
@@ -354,18 +349,42 @@ class CDROM_Interface_Image final : public CDROM_Interface
 };
 
 #if defined (LINUX)
-class CDROM_Interface_Ioctl : public CDROM_Interface_SDL
-{
+class CDROM_Interface_Ioctl : public CDROM_Interface {
 public:
-	CDROM_Interface_Ioctl		();
-
-	bool	SetDevice		(const char* path, const int cd_number);
-	bool	GetUPC			(unsigned char& attr, char* upc);
-	bool	ReadSectors		(PhysPt buffer, const bool raw, const uint32_t sector, const uint16_t num);
-	bool	ReadSectorsHost	(void* buffer, bool raw, unsigned long sector, unsigned long num);
+	~CDROM_Interface_Ioctl();
+
+	bool SetDevice(const char* path, const int cd_number);
+	bool GetUPC(unsigned char& attr, char* upc);
+	bool GetAudioTracks(uint8_t& stTrack, uint8_t& end, TMSF& leadOut);
+	bool GetAudioTrackInfo(uint8_t track, TMSF& start, unsigned char& attr);
+	bool GetAudioSub(unsigned char& attr, unsigned char& track,
+	                 unsigned char& index, TMSF& relPos, TMSF& absPos);
+	bool GetAudioStatus(bool& playing, bool& pause);
+	bool GetMediaTrayStatus(bool& mediaPresent, bool& mediaChanged,
+	                        bool& trayOpen);
+	bool ReadSectors(PhysPt buffer, const bool raw, const uint32_t sector,
+	                 const uint16_t num);
+	bool ReadSectorsHost(void* buffer, bool raw, unsigned long sector,
+	                     unsigned long num);
+	bool PlayAudioSector(const uint32_t start, uint32_t len);
+	bool PauseAudio(bool resume);
+	bool StopAudio();
+	void ChannelControl(TCtrl ctrl);
+	bool LoadUnloadMedia(bool unload);
 
 private:
-	char	device_name[512];
+	void CDAudioCallback(uint16_t requested_frames);
+	void InitAudio();
+
+	int cdrom_fd                            = -1;
+	mixer_channel_t mixer_channel           = nullptr;
+	std::unique_ptr<int16_t[]> audio_buffer = nullptr;
```

Comment:
It would be better to use unsigned unless we really need signed, so `uint16_t`.

Reply:
Well, the data is actually 16 bit signed integers because that's how CD audio is stored.  As of now, I'm not directly manipulating it so it could technically be anything (though it does need to be explicitly cast to `int16_t` before passing the pointer off to `AddSamples_s16`)

Reply:
Ah right, then it makes sense to use `int16_t.